config/routes.rb

```diff
@@ -98,7 +98,9 @@
     post 'registrations/add' => 'registrations#do_add', as: :registrations_do_add
     get 'registrations/psych-sheet' => 'registrations#psych_sheet', as: :psych_sheet
     get 'registrations/psych-sheet/:event_id' => 'registrations#psych_sheet_event', as: :psych_sheet_event
-    resources :registrations, only: [:index, :update, :create, :edit, :destroy], shallow: true
+    resources :registrations, only: [:index, :update, :create, :edit, :destroy], shallow: true do
+      get ':registration_id/payments' => 'payment#registration_payments', as: :payments
```

Comment:
Sorry for picking so much on your `routes.rb`, but the fact that it's nested under `resources :registrations` adds a `:registration_id` route param automagically. `get 'payments'` is literally enough to achieve what you want.